packages/schematics/angular/utility/dependency.ts

```diff
@@ -0,0 +1,78 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import { Rule, SchematicContext } from '@angular-devkit/schematics';
+import { NodePackageInstallTask } from '@angular-devkit/schematics/tasks';
+import * as path from 'path';
+
+const installTasks = new WeakMap<SchematicContext, string[]>();
+
+interface MinimalPackageManifest {
+  dependencies?: Record<string, string>;
+  devDependencies?: Record<string, string>;
+  peerDependencies?: Record<string, string>;
+}
+
+/**
+ * Adds a package as a dependency to a `package.json`. By default the `package.json` located
+ * at the schematic's root will be used. The `manifestPath` option can be used to explicitly specify
+ * a `package.json` in different location. The type of the dependency can also be specified instead
+ * of the default of the `dependencies` section by using the `type` option with either `dev` or `peer`
```

Comment:
Nit: What about optional variants of dependencies? Do we want to support those?

Reply:
Considered them as well but they are not particularly common and could always be added in the future if there is demand.  There would also be optional peer dependencies but those need extra handling since it would need the addition of a `peerDependenciesMeta ` section with the package name.

Reply:
+1 to deferring optional stuff until there's demand.